src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/Publisher.java

```diff
@@ -50,22 +53,38 @@ public Publisher(int evalIntervalSeconds, Collator collator) {
     this.coolOffDetector = new CoolOffDetector();
     // TODO please bring in guice so we can configure this with DI
     this.flipFlopDetector = new TimedFlipFlopDetector(1, TimeUnit.HOURS);
+    this.clock = Clock.systemUTC();
   }
 
   @Override
-  public EmptyFlowUnit operate() {
+  public String name() {
+    return NAME;
+  }
+
+  @Override
+  public PublisherEventFlowUnit operate() {
     // TODO: Need to add dampening, avoidance, state persistence etc.
     Decision decision = collator.getFlowUnits().get(0);
+    List<PersistableAction> persistableActions = new ArrayList<>();
     for (Action action : decision.getActions()) {
       if (coolOffDetector.isCooledOff(action) && !flipFlopDetector.isFlipFlop(action)) {
         flipFlopDetector.recordAction(action);
         coolOffDetector.recordAction(action);
         for (ActionListener listener : actionListeners) {
           listener.actionPublished(action);
         }
+        if (action instanceof PersistableAction) {
+          persistableActions.add((PersistableAction) action);
+        }
```

Comment:
As mentioned above, let's create a `Plugin` that will implement `ActionListener` to persist actions to sql. Then we'll add it to `actionListeners` above and this will get handled implicitly.